src/screens/SubjectsIndex/SubjectsPageContainer.js

```diff
@@ -1,14 +1,24 @@
-import React, { Component } from 'react'
+import React from 'react'
 import { Box } from 'grommet'
+import AppContext from 'store'
+import { generatePath, withRouter } from 'react-router-dom'
+import { EDIT_PATH } from 'paths'
 import ResourcesTable from '../../components/ResourcesTable'
 import { mockColumns, mockData } from './mock'
 
-export default class WorkflowPageContainer extends Component {
-  render() {
-    return (
-      <Box margin='medium' fill='vertical'>
-        <ResourcesTable columns={mockColumns} data={mockData} />
-      </Box>
-    )
+function SubjectsPageContainer (props) {
+  const store = React.useContext(AppContext)
+  const onSelection = (id) => {
+    store.subject.fetchSubject(id, true)
```

Comment:
`fetchSubject()` is called in two areas:
1. on the Subject Viewer/Editor page (showing a single Subject, i.e. http://localhost:3000/projects/123/workflows/123/subject-sets/123/subjects/458050/edit)
2. on the Subjects list (listing all Subjects in a Subject Set, i.e. `http://localhost:3000/projects/123/workflows/123/subject-sets/123/subjects/`), when a Subject is clicked

Question: why is it needed in two locations? Shouldn't it be enough that the Subject Viewer component handle the responsibility of fetching a Subject, when the component is loaded with a specific ID?

Counter argument: but it works.

Counter counter argument: but you're basically duplicating what's essentially an atomic function over two different components, so when you're testing "Is the fetch Subject function working?" you'll need to consider two scenarios: 1. when the user accesses the Subject Viewer/Editor page directly, and 2. when the user accesses the Subject Viewer/Editor by clicking on the Subjects list.

Reply:
That's a good point. I guess it would be more effective to only call `fetchSubject` when the editor is loaded with the given url param. Although it does work, I think you're right that this sort of code duplication is unnecessary.